### PR TITLE
Fix: Fixed an issue with admin operations not working

### LINKED ISSUE #13229: Bug: Files fails all admin oprations
### Description

It will give the prompt to use admin then if given it will still fail.

### Steps To Reproduce

1. Copy a folder/file
2. Paste it into C:\Program Files
3. Give admin access on prompt
4. See copy failed

### Requirements

- Use admin when given

### Files Version

2.5.22.0

### Windows Version

10.0.22621.2134

### Log File

[debug.log](https://github.com/files-community/Files/files/12398301/debug.log)

### PR DESCRIPTION
**Pull request by gave92**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Closes #13229, #10422

[ShellFileOperations2.cs](https://github.com/files-community/Files/pull/13545/files#diff-a27a4cbb71c7049175789aca758b508ff8bbf96650318e27c6177948b73b0a88) is the same file added for #13359 with a small change.

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [ ] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [ ] Did you implement any design changes to an existing feature?
   - [ ] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Copy, delete, rename, etc files from a protected folder

### CODE REVIEW

**Comment by hishitetsu** [comment]
It doesn't show progress, but I still think it is very nice!
Here are a few things I noticed.
* It would be better to show progress as indeterminate.
* Occasionally the following exception occurs:
```
2023-10-20 08:56:19.6985|Error|System.ArgumentException: Source object type is not a projected type and does not inherit from a projected type. (Parameter 'value')
   at WinRT.CastExtensions.GetRefForObject(Object value)
   at WinRT.CastExtensions.As[TInterface](Object value)
   at Files.App.Dialogs.ElevateConfirmDialog.ElevateConfirmDialog_obj3_Bindings.SetDataRoot(Object newDataRoot)
   at Files.App.Dialogs.ElevateConfirmDialog.ElevateConfirmDialog_obj3_Bindings.DataContextChangedHandler(FrameworkElement sender, DataContextChangedEventArgs args)
   at ABI.Windows.Foundation.TypedEventHandler`2.Do_Abi_Invoke[TSenderAbi,TResultAbi](Void* thisPtr, TSenderAbi sender, TResultAbi args)
```

File: src/Files.App/Utils/Shell/ShellFileOperations2.cs [hunk 1]
```diff
@@ -0,0 +1,744 @@
...
+	private OperationFlags opFlags = defaultOptions;
+	private HWND owner;
+	private readonly IFileOperationProgressSink sink;
+	private readonly uint sinkCookie;
+
+	/// <summary>Initializes a new instance of the <see cref="ShellFileOperations"/> class.</summary>
+	/// <param name="owner">The window that owns the modal dialog. This value can be <see langword="null"/>.</param>
+	public ShellFileOperations2(HWND owner = default)
+	{
+		Ole32.CoCreateInstance(typeof(CFileOperations).GUID, null, Ole32.CLSCTX.CLSCTX_LOCAL_SERVER, typeof(IFileOperation).GUID, out var opObj);
```
Comment (gave92) [inline src/Files.App/Utils/Shell/ShellFileOperations2.cs:25]: The trick is using CLSCTX_LOCAL_SERVER, CLSCTX_INPROC_SERVER does not seem to work, meh.

File: src/Files.App/Utils/Shell/ShellFileOperations2.cs
Comment (yaira2) [inline src/Files.App/Utils/Shell/ShellFileOperations2.cs:1]: What does `2` indicate?

Comment (gave92) [reply src/Files.App/Utils/Shell/ShellFileOperations2.cs:1]: ShellFileOperation is already a class from Vanara. This is a copy paste with minor changes.

Comment (yaira2) [reply src/Files.App/Utils/Shell/ShellFileOperations2.cs:1]: Would it be worth highlighting those changes? Or was that already done? 😂

Comment (gave92) [reply src/Files.App/Utils/Shell/ShellFileOperations2.cs:1]: Good idea, will do

**Review by yaira2 — APPROVED** [review]
LGTM

**Comment by gave92** [comment]
One sec "Yes" it's not showing

**Comment by gave92** [comment]
Done! :)

**Review by yaira2 — APPROVED** [review]

**Comment by Lukiluc29** [comment]
I made a copy of several folders on my disk, and when I try to delete them, some of them get deleted, while others ask for administrator permission. When I click 'Yes,' they don't get deleted, and the status center displays an error. Is this normal behavior?

**Pull request merged**